lib/ecto/repo/schema.ex

```diff
@@ -31,7 +31,7 @@ defmodule Ecto.Repo.Schema do
     end
   end
 
-  defp do_insert_all(name, schema, prefix, source, rows, opts) when is_list(rows) do
+  defp do_insert_all(name, schema, prefix, source, rows, opts) when is_list(rows) or is_struct(rows, Ecto.Query) do
```

Comment:
Is there a way to have this when clause in elixir 1.8 at all? Otherwise I'd say we should remove the when clause entirely

Reply:
Not possible. You can validate the arguments separately and raise accordingly.

Reply:
Latest push addresses this